cli_tools/diagnostics/main.go

```diff
@@ -0,0 +1,172 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//	  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"archive/zip"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	tmpFolder   = ""
+	errNonFatal = errors.New("method succeeded with errors")
+)
+
+type runner interface {
+	run() (string, error)
+}
+
+type logFolder struct {
+	name  string
+	files []string
+}
+
+func zipFiles(logs []logFolder, outputPath string) error {
+	newFile, err := os.Create(outputPath)
+	if err != nil {
+		return err
+	}
+	defer newFile.Close()
+	writer := zip.NewWriter(newFile)
+	defer writer.Close()
```

Comment:
You should catch the errors at close (here and for newFile). You can do this using named returns:

func zipFiles(logs []logFolder, outputPath string) (err error) {
...
defer func() {
	cErr := writer.Close()
	if cErr != nil && err == nil {
		err = cErr
	}
}()
}

Reply:
Updated everywhere I close a file.